Backend/app/dashboard/CommonAPIs/src/Dashboard/ProviderPlatform/Driver.hs

```diff
@@ -296,8 +296,34 @@ type DisableDriverAPI =
 type BlockDriverAPI =
   Capture "driverId" (Id Driver)
     :> "block"
+    :> ReqBody '[JSON] BlockDriverReq
     :> Post '[JSON] APISuccess
 
+data BlockDriverReq = BlockDriverReq
+  { code :: Text,
+    blockReason :: Maybe Text,
```

Comment:
This code should have reason code type, right?

Reply:
ok